web/pages/[[...slug]].tsx

```diff
@@ -241,11 +242,25 @@ const Route = ({ data: initialData, slug, preview }: RouteProps) => {
   );
 };
 
-export const getServerSideProps: GetServerSideProps = async ({
-  params,
+export const getStaticPaths: GetStaticPaths = async () => {
+  const allSlugsQuery = groq`*[defined(slug.current) && _type == 'route'][].slug.current`;
+  const pages = await client.fetch(allSlugsQuery);
+  const paths = pages.map((slug) => ({
+    params: {
+      slug: [urlJoin(slug, { leadingSlash: false })],
+    },
+  }));
+
+  return { paths, fallback: 'blocking' };
+};
+
+export const getStaticProps: GetStaticProps = async ({
+  params: { slug: s = '/' },
```

Comment:
This `s` being either an array or a string is confusing for me too. I am not sure, but I suspect that it is an array for nested paths handled by catch-all routes, so that e.g. a _/page/hello/world_ route handled by _/page/[...slug].tsx_ would receive `['hello', 'world']`, but have not verified this assumption.

> Or even leave the default off, because Array.isArray(undefined) would then be false and the '/' could be specified in the "else" branch of the ternary below instead.

I did so in eea44e5, (along with the parameter rename -- as `slug` could be both a string and an array, naming it `slugArray` seemed misleading) but I don't really see how this is much different; I've simply moved the defaulting to `/` in the parameter down. Am I missing something?

Reply:
> This s being either an array or a string is confusing for me too.

@j0nas But I thought the only reason it could be a string is that you made `s` default to a string. Have you ever seen it be a string in any other cases? The Next.js documentation seems to say that it is an array of strings (or `undefined`).

Reply:
I was coding based on the type declaration. Can't remember whether I've personally encountered the case.

![image](https://user-images.githubusercontent.com/6001131/157418295-9708821d-26b3-43cd-a391-1d6fa144f96a.png)

Reply:
OK, I see.

According to the documentation it'll always be an array for `web/pages/[[...slug]].tsx` (so, basically, `string[]?`). The `string` case only applies for non-catch-all routes.

I just found the mismatch between `getStaticPaths` output and `getStaticProps` input a bit confusing, but it's fine to leave it like this.